src/components/App.tsx

```diff
@@ -107,9 +107,13 @@ import { actionFinalize } from "../actions";
 
 function withBatchedUpdates<
   TFunction extends ((event: any) => void) | (() => void)
->(func: TFunction) {
+>(
+  func: Parameters<TFunction>["length"] extends 0 | 1
+    ? TFunction
+    : "max 1 arity function allowed",
+) {
   return (event => {
-    unstable_batchedUpdates(func, event);
+    unstable_batchedUpdates(func as TFunction, event);
   }) as TFunction;
 }
```

Comment:
Could this use overloads instead?

```ts
function withBatchedUpdates(func: () => void): () => void
function withBatchedUpdates<T>(func: (event: T) => void): (event: T) => void
function withBatchedUpdates(func: (event: any) => void) {
  return (event: any) => unstable_batchedUpdates(func, event);
}
```

Reply:
That would still allow passing `(ev: any, history: boolean = true)` functions to it, which my code aims to prevent. Anyway, I'm thinking of going with `never` after all, and just using a JSDoc comment explaining what you're supposed to pass, because it turns out that supplying a `(ev: any, history: boolean)` doesn't show the explaining comment in the TS error message, because it falls into a different type check branch.